### DISCUSSION TITLE
Why does `adoptStyles` re-assign the list? shouldn't it spread any existing adopted stylesheets in?

### DISCUSSION THREAD

**Discussion by bennypowers** [lit/lit#2517] [Ideas]
While working on a "StylesController" which adds CSSResults to an element, I imported `adoptStyles` from 'lit' to use that in my implementation, but that broke chrome-based browsers.

I looked at the code and found that `adoptStyles` clobbers the list. Even [Google's own docs](https://developers.google.com/web/updates/2019/02/constructable-stylesheets#using_constructed_stylesheets) on the feature recommend appending to the list, so how come lit's version doesn't?

```js
const sheet = new CSSStyleSheet();
sheet.replaceSync('a { color: red; }');

// Combine existing sheets with our new one:
document.adoptedStyleSheets = [...document.adoptedStyleSheets, sheet];
```

Lit's version:
https://github.com/lit/lit/blob/7453e365000e6a289c139cf7e175a4742296333d/packages/reactive-element/src/css-tag.ts#L143-L145

My working version:

```ts
  private adoptStyles(): void {
    if (!(this.host.renderRoot instanceof ShadowRoot)) {
      return;
    }

    const styles = [this.styles].flatMap(x => getCompatibleStyle(x as CSSResultOrNative)).filter(x => !!x);

    if (supportsAdoptingStyleSheets) {
      this.host.renderRoot.adoptedStyleSheets = [
        ...this.host.renderRoot.adoptedStyleSheets ?? [],
        ...styles.map(x => x instanceof CSSStyleSheet ? x : x.styleSheet as CSSStyleSheet),
      ];
    } else {
      styles.forEach(s => {
        const style = document.createElement('style');
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const nonce = (window as any)['litNonce'];
        if (nonce !== undefined) {
          style.setAttribute('nonce', nonce);
        }
        style.textContent = (s as CSSResult).cssText;
        this.host.renderRoot.appendChild(style);
      });
    }
    this.stylesAdopted = true;
  }
```

**Comment by sorvell** [comment]
Yeah, I think this is a bit of a rough edge. Calling `adoptStyles` > 1x per element is not something we've focused on, but I think it's reasonable for extensions like what you're doing. As it stands, it's probably easiest to manage the list of styles via `elementStyles/finalizeStyles` if possible and always pass that to `adoptStyles`.

It looks like there's inconsistency between how styles are handled when `adoptedStylesheets` is available (styles are replaced) v. not (styles are added) so there's a bug here. We'll have to think about the least breaking way to address that but it'd be great to file a separate issue for that.
